test/libevmasm/Optimiser.cpp

```diff
@@ -1467,6 +1478,91 @@ BOOST_AUTO_TEST_CASE(cse_replace_too_large_shift)
 	});
 }
 
+BOOST_AUTO_TEST_CASE(inliner)
+{
+	AssemblyItem jumpInto{Instruction::JUMP};
+	jumpInto.setJumpType(AssemblyItem::JumpType::IntoFunction);
+	AssemblyItem jumpOutOf{Instruction::JUMP};
+	jumpOutOf.setJumpType(AssemblyItem::JumpType::OutOfFunction);
+	AssemblyItems items{
+		AssemblyItem(PushTag, 1),
+		AssemblyItem(PushTag, 2),
+		jumpInto,
+		AssemblyItem(Tag, 1),
+		Instruction::STOP,
+		AssemblyItem(Tag, 2),
+		Instruction::CALLVALUE,
+		Instruction::SWAP1,
+		jumpOutOf,
+	};
```

Comment:
Note to self: This is

```
Push [Tag] 1
Push [Tag] 2
JUMP
Tag_1
STOP
Tag_2
CALLVALUE
SWAP1
JUMP
```

Reply:
Does this mean the following?

```
1. Jump to tag_2 setting return location to tag_1
2. Issue callvalue, SWAP1, and jump back to return location previously set
3. Stop
```

Reply:
Kind of - not sure I read correctly what you're points are :-).
The first "Push [Tag] 1" pushes the location of tag_1 as return address to the stack. ``Push [Tag] 2`` pushes the location of tag_2 as jump destination. So this will jump to ``Tag_2`` with ``Tag_1`` still on the stack. ``CALLVALUE`` is then just something the imagined function here does, i.e. push the callvalue to the stack (simulating a function return value). ``SWAP1`` rearranges the swap to move the return address to the stack top, then the ``JUMP`` jumps back to ``Tag_1``.